pkg/network/ovn_kubernetes.go

```diff
@@ -532,10 +534,26 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootstrap.BootstrapResult) (renderIPsecMachineConfig, renderIPsecDaemonSet,
 	renderIPsecOVN, renderIPsecHostDaemonSet, renderIPsecContainerizedDaemonSet, renderIPsecDaemonSetAsCreateWaitOnly bool) {
 	isHypershiftHostedCluster := bootstrapResult.Infra.HostedControlPlane != nil
-	isIPsecEnabled := conf.IPsecConfig != nil
 	isIpsecUpgrade := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.LegacyIPsecUpgrade
 	isOVNIPsecActive := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.OVNIPsecActive
 
+	// Find the IPsec mode from Ipsec.config
+	// Ipsec.config == nil (bw compatibility) || ipsecConfig == Off ==> ipsec is disabled
+	// ipsecConfig.mode == "" (bw compatibility) || ipsec.Config == Full ==> ipsec is enabled for NS and EW
+	// ipsecConfig.mode == External ==> ipsec is enabled for NS only
+
+	mode := operv1.IPsecModeDisabled // Should stay so if conf.IPsecConfig == nil
+	if conf.IPsecConfig != nil {
+		if conf.IPsecConfig.Mode != "" {
+			mode = conf.IPsecConfig.Mode
+		} else {
+			mode = operv1.IPsecModeFull // BW compatibility with existing configs
+		}
+	}
+	isIPsecEnabled := mode != operv1.IPsecModeDisabled
+	isIPsecFull := mode == operv1.IPsecModeFull // Full mode is for NS+EW
+	klog.Infof("IPsec: after looking at %+v, ipsec mode=%s, isIPsecEnabled=%v", conf.IPsecConfig, mode, isIPsecEnabled)
```

Comment:
NIT: is there any value in logging the `isIPsecEnabled`? In most deployments `IPsecConfig` will be nil and I am not sure if we need that log message(looks more like a v5?)